pkg/cidata/cidata.TEMPLATE.d/boot/04-persistent-data-volume.sh

```diff
@@ -74,6 +77,19 @@ if [ "$(awk '$2 == "/" {print $3}' /proc/mounts)" == "tmpfs" ]; then
 				for DIR in ${DATADIRS}; do
 					DEST="/mnt/data$(dirname "${DIR}")"
 					mkdir -p "${DIR}" "${DEST}"
+					# Unmount filesystems (virtiofs, 9p, etc.) below "${DIR}", otherwise host mounts will be wiped out
+					# https://github.com/rancher-sandbox/rancher-desktop/issues/6582
+					#
+					# set is for avoiding a failure on empty grep
+					set +e +o pipefail
+					awk '$3 != "ext4" && $3 != "tmpfs" {print $2}' /proc/mounts | grep "^${DIR}" | while IFS= read -r line; do
+						[ -z "$line" ] && continue
+						# unmangle " \t\n\\#"
+						# https://github.com/torvalds/linux/blob/v6.6/fs/proc_namespace.c#L89
+						unmangled="$(echo "$line" | sed -e 's/\\040/ /g; s/\\011/\t/g; s/\\012/\n/g; s/\\134/\\/g; s/\\043/#/g')"
+						umount "$unmangled" || exit 1
```

Comment:
Should this be a `break` instead of `exit 1`?

Reply:
exit 1 might be safer?

Reply:
Maybe. I was thinking if one directory cannot be unmounted because the device is busy, then we could just ignore moving this one directory, but still move the others.

But either way, it is surprising behaviour unless you actually check the logs.